### PR TITLE
Simplify array_replace description and add nested array example

### PR DESCRIPTION
**Pull request by jimwins**
Based on discussion at PR #1470

### CODE REVIEW

File: reference/array/functions/array-replace.xml [hunk 1]
```diff
@@ -85,6 +80,40 @@ Array
...
+    <title>Example of how nested arrays are handled</title>
+    <programlisting role="php">
+<![CDATA[
+<?php
+$base = [ 'citrus' => [ 'orange', 'lemon' ], 'pome' => [ 'apple' ] ];
+$replacements = [ 'citrus' => [ 'grapefruit' ] ];
+$replacements2 = [ 'citrus' => [ 'kumquat', 'citron' ], 'pome' => [ 'loquat' ] ];
+
+$basket = array_replace($base, $replacements, $replacements2);
+print_r($basket);
```
Comment (Girgias) [inline reference/array/functions/array-replace.xml:96]: I usually prefer `var_dump()` then `print_r()` as it is whitespace sensitive in very awkward places :/

Comment (jimwins) [reply reference/array/functions/array-replace.xml]: I just copied what the first example was doing. I've changed them both to `var_dump()`.

Comment (jimwins) [reply reference/array/functions/array-replace.xml]: I did leave alone the `array(...)` syntax on the first example even though I used `[ ... ]` in the second.

**Review by Girgias — APPROVED** [review]
Thank you!

**Pull request merged**